### PR TITLE
[Simplify Interface]add detach

### PR DESCRIPTION
**Pull request by qicosmos**
## Why
simplify code:
```via(&executor).start([](auto&&) {});```
 to
```via(&executor).detach();```

### CODE REVIEW

File: async_simple/coro/Lazy.h [hunk 1]
```diff
@@ -509,6 +509,10 @@ class [[nodiscard]] RescheduleLazy {
             launchCoro(std::move(*this), std::forward<F>(callback));
     }
 
+    void detach() {
+        start([](auto&&) {});
```
Comment (RainMark) [inline async_simple/coro/Lazy.h:513]: in previous issue #17 we know user forget to handle exception with `start([](auto&&) {});`
we have to rethrow the existing exception in `detach` api, because user can not use Try type to handle it any more

```cpp
start([](auto&& t) {
  if (t.hasError()) {
    // rethrow
  }
});
```

Comment (qicosmos) [reply async_simple/coro/Lazy.h]: OK

**Review by ChuanqiXu9 — APPROVED** [review]

**Review by RainMark — APPROVED** [review]
